src/main/kotlin/com/openlattice/chronicle/jobs/DeleteChronicleUsageDataTask.kt

```diff
@@ -0,0 +1,140 @@
+import java.time.OffsetDateTime
+import com.openlattice.chronicle.auditing.AuditingManager
+import com.openlattice.chronicle.ids.IdConstants
+import com.openlattice.chronicle.services.jobs.JobService
+import com.openlattice.chronicle.storage.ChroniclePostgresTables.Companion.JOBS
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.CREATED_AT
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.DELETED_ROWS
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_DATA
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_ID
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.STATUS
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.UPDATED_AT
+import com.openlattice.chronicle.storage.RedshiftColumns.Companion.STUDY_ID
+import com.openlattice.chronicle.storage.RedshiftDataTables.Companion.CHRONICLE_USAGE_EVENTS
+import com.openlattice.chronicle.storage.StorageResolver
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+/**
+ * @author Solomon Tang <solomon@openlattice.com>
+ */
+class DeleteChronicleUsageDataTask(
+    private val jobService: JobService,
+    private val storageResolver: StorageResolver,
+    private val auditingManager: AuditingManager,
+): Runnable {
+    companion object {
+        private val logger = LoggerFactory.getLogger(DeleteChronicleUsageDataTask::class.java)!!
+
+        /*
+          1. status
+          2. updated_at
+         */
+        private val GET_NEXT_JOB_COLUMNS = listOf(
+            UPDATED_AT,
+            STATUS
+        ).joinToString(",") { it.name }
+
+        private val GET_NEXT_JOB_SQL = """
+            UPDATE ${JOBS.name}
+            SET (${GET_NEXT_JOB_COLUMNS}) = (?, ?)
+            WHERE ${JOB_ID.name} = (
+                SELECT ${JOB_ID.name}
+                FROM ${JOBS.name}
+                WHERE ${JOB_DATA.name} ->> '@type' = '${DeleteStudyUsageData::class.java.name}'
+                AND ${STATUS.name} = '${JobStatus.PENDING.name}'
+                ORDER BY ${CREATED_AT.name} ASC
+                FOR UPDATE SKIP LOCKED
+                LIMIT 1
+            )
+            RETURNING *
+        """.trimIndent()
+
+        private val DELETE_CHRONICLE_STUDY_USAGE_DATA_SQL = """
+            DELETE FROM ${CHRONICLE_USAGE_EVENTS.name}
+            WHERE ${STUDY_ID.name} = ?
+        """.trimIndent()
+
+        private  val UPDATE_FINISHED_JOB_COLUMNS = listOf(
+            UPDATED_AT,
+            STATUS,
+            DELETED_ROWS
+        ).joinToString(",") { it.name }
+
+        private val UPDATE_FINISHED_DELETE_JOB_SQL = """
+            UPDATE ${JOBS.name}
+            SET (${UPDATE_FINISHED_JOB_COLUMNS}) = (?, ?, ?)
+            WHERE ${JOB_ID.name} = ?
+        """.trimIndent()
+    }
+
+    override fun run() {
+        deleteChronicleUsageData()
+    }
+
+    private fun deleteChronicleUsageData() {
+        val hds = storageResolver.getPlatformStorage()
```

Comment:
Usage data does not live in platform storage resolve the study.

Reply:
this function is named poorly as its real purpose is to just audit what happens on the platform side of things. 

`deleteChronicleStudyUsageData` down below gets its own connection to using `getDefaultEventStorage()`

i'll update to make this more clear

Reply:
as for resolving the study, at this point the study has already been deleted from the platform side of things.

`destroyStudy` creates the job to delete usage data, deletes the study and it's associations to any organizations.

This task merely picks up the job and cleans up the data in redshift